### ISSUE TITLE
pip install sentencepiece failed (error: ‘Swap32’ is not a member of ‘sentencepiece::util’ .....)

### ISSUE THREAD

**Issue by fulwang** [google/sentencepiece#912]
Hi there,

I'm trying to install this package inside an Anaconda virtual env on Red Hat Linux 8.6 running on IBM system z and didn't find any meaning infor about this issue.

Could anyone have a look and kindly give a suggest?

Thanks!

(base) [root@lpar50 ~]# 
(base) [root@lpar50 ~]# uname -ar
Linux lpar50 4.18.0-372.9.1.el8.s390x #1 SMP Fri Apr 15 22:05:17 EDT 2022 s390x s390x s390x GNU/Linux
(base) [root@lpar50 ~]# 
(base) [root@lpar50 ~]# conda --version
conda 23.7.3
(base) [root@lpar50 ~]# 

Details, pls refer attached log.

[sentencepiece_install.log](https://github.com/google/sentencepiece/files/12605832/sentencepiece_install.log)

**Comment by chenqiny** [comment]
@fulwang  I encountered same thing. Pls try 0.1.94

**Comment by chenqiny** [comment]
Furthur more information:
I can't install it on s390x with pip. But I can compile 0.1.99 with source manually.

Compilation step from source:
cmake .. -DCMAKE_CXX_FLAGS=-march=z196
make -j4

I guess this is cause by build_bundled.sh. I did not find a way to send the arch information to cmake

```
  [ 24%] Building CXX object src/CMakeFiles/sentencepiece-static.dir/util.cc.o
    /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.cc: In static member function 'static std::string sentencepiece::normalizer::Normalizer::EncodePrecompiledCharsMap(std::string_view, std::s
tring_view)':
    /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.cc:268:21: error: 'Swap32' is not a member of 'sentencepiece::util'; did you mean 'Swap32'?
      268 |     data[i] = util::Swap32(data[i]);
          |                     ^~~~~~
    In file included from /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.h:24,
                     from /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.cc:15:
    /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/common.h:83:15: note: 'Swap32' declared here
       83 | inline uint32 Swap32(uint32 x) { return __builtin_bswap32(x); }
          |               ^~~~~~
    /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.cc: In static member function 'static sentencepiece::util::Status sentencepiece::normalizer::Normalizer::DecodePrecompiledCharsMap(std::str
ing_view, std::string_view*, std::string_view*, std::string*)':
    /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.cc:288:26: error: 'Swap32' is not a member of 'sentencepiece::util'; did you mean 'Swap32'?
      288 |   trie_blob_size = util::Swap32(trie_blob_size);
          |                          ^~~~~~
    In file included from /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.h:24,
                     from /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.cc:15:
    /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/common.h:83:15: note: 'Swap32' declared here
       83 | inline uint32 Swap32(uint32 x) { return __builtin_bswap32(x); }
          |               ^~~~~~                                                                                                                                                                                                                         /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.cc:299:64: error: 'Swap32' is not a member of 'sentencepiece::util'; did you mean 'Swap32'?                                                      299 |   for (int i = 0; i < trie_blob_size / 4; ++i) data[i] = util::Swap32(data[i]);                                                                                                                                                                    |                                                                ^~~~~~
    In file included from /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.h:24,
                     from /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/normalizer.cc:15:
    /tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/sentencepiece/src/common.h:83:15: note: 'Swap32' declared here
       83 | inline uint32 Swap32(uint32 x) { return __builtin_bswap32(x); }
          |               ^~~~~~                                                                                                                                                                                                                         [ 25%] Building CXX object src/CMakeFiles/sentencepiece-static.dir/word_model.cc.o
    gmake[2]: *** [src/CMakeFiles/sentencepiece-static.dir/build.make:524: src/CMakeFiles/sentencepiece-static.dir/normalizer.cc.o] Error 1
    gmake[2]: *** Waiting for unfinished jobs....
    gmake[1]: *** [CMakeFiles/Makefile2:228: src/CMakeFiles/sentencepiece-static.dir/all] Error 2
    gmake: *** [Makefile:171: all] Error 2
    Traceback (most recent call last):
      File "<string>", line 1, in <module>                                                                                                                                                                                                                 File "/tmp/pip-install-zlz04jwy/sentencepiece_98ac0312f9f343ad9bbfd88e83d8a61a/setup.py", line 167, in <module>
        setup(
      File "/usr/lib/python3.9/site-packages/setuptools/__init__.py", line 153, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib64/python3.9/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/lib64/python3.9/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/usr/lib64/python3.9/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/usr/lib/python3.9/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/usr/lib64/python3.9/distutils/command/install.py", line 557, in run
        self.run_command('build')
      File "/usr/lib64/python3.9/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib64/python3.9/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/usr/lib64/python3.9/distutils/command/build.py", line 135, in run
        self.run_command(cmd_name)
      File "/usr/lib64/python3.9/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib64/python3.9/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/usr/lib/python3.9/site-packages/setuptools/command/build_ext.py", line 79, in run
        _build_ext.run(self)
      File "/usr/lib64/python3.9/distutils/command/build_ext.py", line 340, in run
        self.build_extensions()
      File "/usr/lib64/python3.9/distutils/command/build_ext.py", line 449, in build_extensions
        self._build_extensions_serial()
      File "/usr/lib64/python3.9/distutils/command/build_ext.py", line 474, in _build_extensions_serial
        self.build_extension(ext)
      File "/tmp/pip-install-0_3jkrco/sentencepiece_bba3d28c4d194a8db83eebb19bb3b7ce/setup.py", line 87, in build_extension
        subprocess.check_call(['./build_bundled.sh', __version__])
      File "/usr/lib64/python3.9/subprocess.py", line 373, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['./build_bundled.sh', '0.1.99']' returned non-zero exit status 2.
```

**Comment by taku910** [comment]
Fixed in v0.2.0

**Comment by djfitzgerald** [comment]
@taku910 Is there a workaround that I can use to get sentencepiece==0.1.99 to build on s390x?  I need it for a dependency that won't take sentencepiece==0.2.0
